### PR TITLE
Cluster commands respecting keyslots from command input

### PR DESCRIPTION
**Pull request by greggjs**
Currently, the cluster command "GETKEYSINSLOT" passes in a keyslot as a parameter. This parameter needs to be recognized as the slot to use when executing the command. Currently, it ignores it and picks a random node. This PR fixes this issue.

### CODE REVIEW

**Comment by vmihailenco** [comment]
Could you instead change https://github.com/go-redis/redis/blob/dba9cb31f566f15c82162dbde38ed8939d8b8e5a/cluster.go#L777-L780 like this:

```
func (c *ClusterClient) cmdSlot(cmd Cmder) int {
        args := cmd.Args()
	if args[0] == "cluster" && args[1] == "getkeysinslot" {
		return args[2].(int)
	}

	cmdInfo := c.cmdInfo(cmd.Name())
	return cmdSlot(cmd, cmdFirstKeyPos(cmd, cmdInfo))
}
```

**Comment by greggjs** [comment]
Yea sure no prob man

File: cluster.go [hunk 1]
```diff
@@ -767,11 +767,19 @@ func cmdSlot(cmd Cmder, pos int) int {
 	if pos == 0 {
 		return hashtag.RandomSlot()
 	}
-	firstKey := cmd.stringArg(pos)
-	return hashtag.Slot(firstKey)
+	val, ok := cmd.Args()[pos].(int)
+	if ok {
+		return val
+	}
+	return hashtag.Slot(cmd.stringArg(pos))
```
Comment (vmihailenco) [inline cluster.go]: Can we revert this change?

File: command.go [hunk 2]
```diff
@@ -74,6 +74,11 @@ func cmdString(cmd Cmder, val interface{}) string {
 
 func cmdFirstKeyPos(cmd Cmder, info *CommandInfo) int {
 	switch cmd.Name() {
+	case "cluster":
+		switch cmd.stringArg(1) {
+		case "getkeysinslot":
+			return 2
+		}
```
Comment (vmihailenco) [inline command.go]: can we revert this change as well?

File: cluster.go (hunk 1 above)
Comment (greggjs) [reply cluster.go]: So if you look at this line of code, https://github.com/go-redis/redis/blob/f3804cd0b74c6515ef206dabe7b314f2765fc788/cluster.go#L794

We reference the code that I changed. Would you suggest I change this method call to the cluster client specific method?

File: command.go (hunk 2 above)
Comment (greggjs) [reply command.go]: Sure

File: cluster.go (hunk 1 above)
Comment (greggjs) [reply cluster.go]: I updated the code to reflect the change I was referring to.

**Comment by vmihailenco** [comment]
Thanks
